test/validation/api/ml/ml.c

```diff
@@ -0,0 +1,501 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2023 Nokia
+ */
+
+#include <odp_api.h>
+#include <odp/helper/odph_api.h>
+#include "odp_cunit_common.h"
+
+#define UAREA     0xaa
+#define NUM_COMPL 10u
+#define COMPL_POOL_NAME "ML compl pool"
+
+typedef struct global_t {
+	int disabled;
+	uint32_t num_compl;
+	odp_ml_capability_t ml_capa;
+} global_t;
+
+typedef struct {
+	uint32_t count;
+	uint8_t mark[NUM_COMPL];
+} uarea_init_t;
+
+static global_t global;
+
+static int ml_suite_init(void)
+{
+	memset(&global, 0, sizeof(global_t));
+
+	if (odp_ml_capability(&global.ml_capa)) {
+		ODPH_ERR("ML capability failed\n");
+		return -1;
+	}
+
+	if (global.ml_capa.max_models == 0) {
+		global.disabled = 1;
+		ODPH_DBG("ML test disabled\n");
+		return 0;
+	}
+
+	global.num_compl = ODPH_MIN(NUM_COMPL, global.ml_capa.pool.max_num);
+
+	return 0;
+}
+
+static int check_ml_support(void)
+{
+	if (global.disabled)
+		return ODP_TEST_INACTIVE;
+
+	return ODP_TEST_ACTIVE;
+}
+
+static void test_ml_capability(void)
```

Comment:
`test_packet_pool_ext_capa()` could be updated to include `ODP_POOL_ML_COMPL` in unsupported types.